control-plane/catalog/to-consul/resource.go

```diff
@@ -547,6 +562,19 @@ func (t *ServiceResource) generateRegistrations(key string) {
 				r.Service.ID = serviceID(r.Service.Service, addr)
 				r.Service.Address = addr
 
+				// Adding information about service weight.
+				// Overrides the existing weight if present
+				if weight, ok := svc.Annotations[annotationServiceWeight]; ok && weight != "" {
```

Comment:
this line 567 - 577 and lines 518 - 528 are exactly the same. can we take it out in a function as well?

Reply:
The code at both place is constructing the CatalogRegistration object, there is logic which is duplicate. Read the comment https://github.com/hashicorp/consul-k8s/pull/2293#discussion_r1221683819